### PR TITLE
Fixes #4467 : Updates time_series for pandas deprecation

### LINKED ISSUE #4467: Update make_timeseries for pandas deprecation
```python
In [3]: dd.demo.make_timeseries()
/Users/taugspurger/sandbox/dask/dask/dataframe/io/demo.py:91: FutureWarning: Creating a DatetimeIndex by passing range endpoints is deprecated.  Use `pandas.date_range` instead.
  freq=partition_freq))
/Users/taugspurger/sandbox/dask/dask/dataframe/io/demo.py:45: FutureWarning: Creating a DatetimeIndex by passing range endpoints is deprecated.  Use `pandas.date_range` instead.
  index = pd.DatetimeIndex(start=start, end=end, freq=freq, name='timestamp')
Out[3]:
Dask DataFrame Structure:
                   id    name        x        y
npartitions=11
2000-01-31      int64  object  float64  float64
2000-02-29        ...     ...      ...      ...
...               ...     ...      ...      ...
2000-11-30        ...     ...      ...      ...
2000-12-31        ...     ...      ...      ...
Dask Name: make-timeseries, 11 tasks
```

Switching to `pd.date_range` should be fine.

### PR DESCRIPTION
**Pull request by HSR05**
Changes pd.DatetimeIndex to pd.date_range

- [x] Tests added / passed
- [X] Passes `flake8 dask`

cc : @mrocklin please review the PR. 
Cheers :+1:

### CODE REVIEW

**Comment by HSR05** [comment]
ref : https://github.com/dask/dask/issues/4467

**Comment by mrocklin** [comment]
cc @TomAugspurger or @jrbourbeau in case you have time to review

File: dask/dataframe/io/demo.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def make_timeseries(start='2000-01-01',
     2000-01-01 06:00:00   960   Charlie  0.788245
     2000-01-01 08:00:00  1031     Kevin  0.466002
     """
-    divisions = list(pd.DatetimeIndex(start=start, end=end,
+    divisions = list(pd.date_range(start=start, end=end,
                                       freq=partition_freq))
```
Comment (TomAugspurger) [inline dask/dataframe/io/demo.py]: This line may fail this linter. Need to decent it a bit.

**Comment by TomAugspurger** [comment]
I see a few more uses of the deprecated pandas behavior throughout dask. Do you mind updating those while you're at it?

```
dask/dataframe/io/tests/test_demo.py
21:    assert df.divisions == tuple(pd.DatetimeIndex(start='2000', end='2015',

dask/dataframe/io/demo.py
45:    index = pd.DatetimeIndex(start=start, end=end, freq=freq, name='timestamp')
90:    divisions = list(pd.DatetimeIndex(start=start, end=end,

dask/dataframe/tests/test_dataframe.py
1393:    ind = pd.DatetimeIndex(start=start, end=end, freq='60s')

dask/dataframe/core.py
4296:    divisions = pd.DatetimeIndex(start=start,
```


```
dask/dataframe/utils.py
393:        return pd.PeriodIndex(start='1970-01-01', periods=2, freq=idx.freq,
```

**Comment by HSR05** [comment]
@TomAugspurger yeah sure. I'll look into those and update it asap. 👍

**Comment by HSR05** [comment]
@TomAugspurger updated the deprecated pandas behavior throughout dask. 
Please review the PR.

Cheers 👍

File: dask/dataframe/utils.py [hunk 2]
```diff
@@ -390,7 +390,7 @@ def _nonempty_index(idx):
         return pd.DatetimeIndex(data, start=start, periods=2, freq=idx.freq,
                                 tz=idx.tz, name=idx.name)
     elif typ is pd.PeriodIndex:
-        return pd.PeriodIndex(start='1970-01-01', periods=2, freq=idx.freq,
+        return pd.date_range(start='1970-01-01', periods=2, freq=idx.freq,
```
Comment (TomAugspurger) [inline dask/dataframe/utils.py]: This should be `pd.period_range`.

Comment (HSR05) [reply dask/dataframe/utils.py]: > 
> 
> This should be `pd.period_range`.

My bad. Thanks for the review. 
Updated it. @TomAugspurger

**Comment by TomAugspurger** [comment]
Some linting errors: https://travis-ci.org/dask/dask/jobs/499881647#L2092

you can run `flake8` locally on those files to verify.

**Comment by mrocklin** [comment]
I've pushed flake8 fixes.  The current failure is unrelated, and being handled in #4537 

@TomAugspurger is everything else here good to be merged?

**Comment by TomAugspurger** [comment]
👍

**Comment by TomAugspurger** [comment]
Thanks @HSR05!

**Comment by mrocklin** [comment]
Thanks @HSR05 !

Also, I notice that this is your first code contribution to this repository.  Welcome!

**Pull request merged**